### PR TITLE
one more (without ebidences but with disemvowel and ooc-mute)

### PR DESCRIPTION
**Pull request by shogunator1337**
Reworked evidences, added iniswap-blocking, reworked targeting-system, ips and hdids united under unique 'ipid' which improving an anonimity without worsening of moder's privileges (all useless ip and hdid stuff was removed same as rp-mode)

### CODE REVIEW

File: server/masterserverclient.py [hunk 1]
```diff
@@ -42,23 +42,38 @@ def __init__(self, server):
...
+                for data in data_list:
+                    raw_msg = data.decode()
+                    cmd, *args = raw_msg.split('#')
+                    if cmd != 'CHECK' and cmd != 'PONG':
+                        logger.log_debug('[MASTERSERVER][INC][RAW]{}'.format(raw_msg))
+                    if cmd == 'CHECK':
+                        await self.send_raw_message('PING#%')
+                    if cmd == 'PONG':
+                        fl = False
+                    elif cmd == b'NOSERV':
```
Comment (oldmud0) [inline server/masterserverclient.py]: wait so why is this one prefixed with `b` and not the other ones

File: server/client_manager.py [hunk 2]
```diff
@@ -67,12 +86,36 @@ def change_character(self, char_id, force=False):
                 raise ClientError('Invalid Character ID.')
             if not force and not self.area.is_char_available(char_id):
                 raise ClientError('Character not available.')
+            elif not self.area.is_char_available(char_id):
```
Comment (oldmud0) [inline server/client_manager.py]: probably might want to wind this if block differently:
```python
if not self.area.is_char_available(char_id):
  if force:
    # Kick everyone else off that character
    # ...
  else:
    raise ClientError('Character not available.')

File: config_sample/areas.yaml [hunk 3]
```diff
@@ -1,9 +1,26 @@
 - area: Basement
   background: gs4
   bglock: true
+  evidence_mod: FFA
+  locking_allowed: false
+  iniswap_allowed: true
```
Comment (oldmud0) [inline config_sample/areas.yaml:6]: will old configs work, even without these new settings?

**Review by oldmud0 — APPROVED** [review]
this code base has gotten pretty ugly. but you should start testing this to see if it works before I merge

File: config_sample/areas.yaml (hunk 3 above)
Comment (shogunator1337) [reply config_sample/areas.yaml:6]: sure

File: server/client_manager.py (hunk 2 above)
Comment (shogunator1337) [reply server/client_manager.py]: fairly

**Comment by shogunator1337** [comment]
What do you mean by 'ugly'? I believe that it's more beatiful than before. But i should add some get_evi_id() (instead of evi_list[client.evi_list[..]]) stuff for sure. 8 people (average) tested this version, it works fine (we tested it for 2 days, i fixed everything we found (thats why here so much commits actually)).

HiddenCM works like evidences on serverD before but with realtime-editing (present is presenting evidence for everyone).

**Comment by shogunator1337** [comment]
here it is.

**Comment by shogunator1337** [comment]
Tested all targetings, HiddenCM works fine (here a problem that if u didn't ever wrote something after logging then ur pos is '' and u didn't see any evidence but here 0 chances to fix it except making default pos for every char), Mods works fine (by now), FFA and CM worked before. Iniswap-blocking worked laggy but i fixed it yesterday so it works now. Music flood-guard works ok, disabled if config is outdated. Locks for rooms and inviting-system works ok. Here were no way for player to break it, but now it's hard to break for server-owner too :^)

This build is started on FOC so here a live-test anyways 🤔 

If here will be 0 commits for 24 hours then it's stable build i believe.

**Comment by oldmud0** [comment]
ugly, as in, the combined changes of everyone have caused a necessity for a little bit of refactoring

File: server/aoprotocol.py [hunk 4]
```diff
@@ -414,7 +414,7 @@ def net_cmd_mc(self, args):
                 self.client.send_host_message("You have been muted by a moderator")
                 return
             if not self.client.is_dj:
-                self.client.send_host_message('You are was undj\'d by a moderator.')
+                self.client.send_host_message('You are was blockdj\'d by a moderator.')
```
Comment (oldmud0) [inline server/aoprotocol.py]: `You were`*

File: README.md [hunk 5]
```diff
@@ -108,12 +110,20 @@ Install PyYAML by doing
...
-    - Changes evidence_mod in this area. Possible values: FFA, CM. In work: HiddenCM, Mod
+    - Changes evidence_mod in this area. Possible values: FFA, CM, HiddenCM, Mods
+        * **FFA**
+            - Everyone can add, edit and remove evidence.
+        * **Mods**
+            - Only moderators can add, edit or remove evidence.
+        * **CM**
+            - Only CM (case-maker, look at /cm for more info) or moderators can add, edit or remove evidence.
+        * **HiddenCM**
+            - Same as CM, but every evidence have his "owner's position" that can be changed by CM or moderator, what means that only one side can see it (except for 'all'). After presenting evidence once his position changing to 'all'. Possible positions: 'def', 'pro', 'wit', 'jud', 'pos' (means that no one can see this evidence), 'all' (means that everyone can see this evidence).
```
Comment (oldmud0) [inline README.md]: Not a very grammatically correct explanation. Try this:

> **HiddenCM**: Same as CM, but every evidence has a preset "owner's position" which can be set by a CM or moderator, such that only one side/position of the court may see the evidence. After presenting the evidence, the position of the evidence changes to "all." Possible positions include `def` (defense), `pro` (prosecutor), `wit` (witness), `jud` (judge), `pos` (hidden from everyone), and `all` (everyone can see the evidence).

Comment (shogunator1337) [reply README.md]: thx, fixed

**Pull request merged**